### PR TITLE
Part-2: Compiler Warnings as Errors - WordPress Module - FragmentManager

### LINKED ISSUE #17253: Compiler Warnings as Errors - WordPress Module - FragmentManager
Parent #17173

This issue is about resolving the `getFragmentManager(...)` and `requireFragmentManager()` compile warnings for the WordPress module, and possibly, for the lib modules as well.

PS: Some of those warnings are already deprecated (see [here](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/75db65bced75dfbd53b0c34797eafa4e4024f6b9)).

-----

- This `getFragmentManager(...)` method is deprecated. This has been removed in favor of `getParentFragmentManager()` which throws an `IllegalStateException` if the `FragmentManager` is null. Check if `isAdded` returns false to determine if the `FragmentManager` is null.
- This `requireFragmentManager(...)` method is deprecated. This has been renamed to `getParentFragmentManager()` to make it clear that you are accessing the `FragmentManager` that contains this Fragment and not the `FragmentManager` associated with child Fragments.

-----

For more info see:
- [Fragment.getFragmentManager](https://developer.android.com/reference/androidx/fragment/app/Fragment#getFragmentManager())
- [Fragment#.requireFragmentManager](https://developer.android.com/reference/androidx/fragment/app/Fragment#requireFragmentManager())

### PR DESCRIPTION
**Pull request by 07jasjeet**
# ONLY merge this after [Part 1](https://github.com/wordpress-mobile/WordPress-Android/pull/19471) has been merged! (Derivative branch)

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/17253 partially. (Sub-ticket of https://github.com/wordpress-mobile/WordPress-Android/pull/17246)

To Test:

- [AddCategoryFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/posts/AddCategoryFragment.java): 

   1) Have a site with a post.
   2) Navigate to "My Site" and click on "Posts" to see all your posts. Open any post in edit mode.
   3) Click on dropdown button on top-left.
   4) Click "Post settings".
   5) Select/Open "Categories".
   6) Click on "+" symbol on top left.
   7) A dialog should open without errors.
- [CollapseFullScreenDialogFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/CollapseFullScreenDialogFragment.java):

   1. Navigate to "Reader" tab.
   2. On any post, click on "Comment" to open comments.
   3. On the bottom of the screen, you will have the text field to reply. Beside that, you will have an "Arrow up" button. Click it.
   4. This is our full screen dialog that got changed. It should work as expected.
- [SignupEpilogueFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java):

   1. Just follow the **signup** flow till you reach a form where you have to fill your name, display name and password.
   2. This is hard to test (and I need help here). The changes in this file are minute and are certainly not going to break anything afaik.

- [SitePickerActivity.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java):

  1. Navigate to "My Site" tab.
  2. Click on "Add Site" button if you have no sites else, click on current opened site's title on top of the page (with an arrow).
  3. You will have "Choose site" screen infront of you.
  4. Click on "+" button on top right. (on Action bar)
  5. Dialog should work as expected.

- [PersonDetailFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/people/PersonDetailFragment.java):

  1. Have at least one site.
  2. Navigate to "My Site" tab.
  3. Click on "More".
  4. Click on "People"
  5. Click on any person.
  6. A detailed view of the person should pop up as expected.

## Regression Notes
1. Potential unintended areas of impact
   Mostly none because the newer API's are a direct replacement.

7. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual Testing

8. What automated tests I added (or what prevented me from doing so)
   None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by 07jasjeet** [comment]
@antonis Please add yourself as a reviewer.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java [hunk 1]
```diff
@@ -336,7 +336,7 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
                 mImageManager.loadIntoCircle(mHeaderAvatar, ImageType.AVATAR_WITHOUT_BACKGROUND, mPhotoUrl);
             }
         } else {
-            mDialog = (FullScreenDialogFragment) getFragmentManager().findFragmentByTag(FullScreenDialogFragment.TAG);
+            mDialog = (FullScreenDialogFragment) getParentFragmentManager().findFragmentByTag(FullScreenDialogFragment.TAG);
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java]: I've noticed a minor checkstyle issue on this line:
```
Line is longer than 120 characters (found 124). [LineLength]
```

Comment (07jasjeet) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java]: Oops looks like the checkstyle error might have went out of screen as well haha. Will fix it!

Comment (07jasjeet) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java]: Also i will be leaving test directions soon so I would request you to wait on this one until then.

Comment (07jasjeet) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java]: @antonis test instructions are up!

Comment (antonis) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.java]: Thank you for the heads up @07jasjeet 🙇 
My plan is to review the PR later today or tomorrow.

File: WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java [hunk 2]
```diff
@@ -833,36 +833,36 @@ public static void addSite(Activity activity, boolean hasAccessToken, SiteCreati
...
      */
-    @SuppressWarnings("deprecation")
     public static class AddSiteDialog extends DialogFragment {
         static final String ADD_SITE_DIALOG_TAG = "add_site_dialog";
 
         @NonNull
         @Override
-        @SuppressWarnings("deprecation")
         public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+            assert getArguments() != null;
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java]: Thank you for adding this extra check @07jasjeet 👍 
What would you say on replacing it with `requireArguments();`

(thank you @ParaskP7 for the suggestion 🙇)

File: WordPress/src/main/java/org/wordpress/android/ui/people/PersonDetailFragment.java [hunk 3]
```diff
@@ -83,9 +83,10 @@ public static PersonDetailFragment newInstance(long currentUserId, long personId
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        ((WordPress) getActivity().getApplicationContext()).component().inject(this);
+        ((WordPress) requireActivity().getApplicationContext()).component().inject(this);
 
         if (savedInstanceState == null) {
+            assert getArguments() != null;
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/people/PersonDetailFragment.java]: As above https://github.com/wordpress-mobile/WordPress-Android/pull/19490/files#r1384759546

**Review by antonis — APPROVED** [review]
Thank you for your work on this @07jasjeet and for handling the review feedback 🙇 
I've tested the implementation and everything worked as expected for me:
- [x] AddCategoryFragment.java
- [x] CollapseFullScreenDialogFragment.java
- [x] SignupEpilogueFragment.java
- [x] SitePickerActivity.java
- [x] PersonDetailFragment.java
- [x] FullScreenDialogFragment.java

The code also looks good 🎉

**Pull request merged**